### PR TITLE
add lib.es2018.intl + fix lib.es2017.intl

### LINKED ISSUE #23691: Add Intl.PluralRules to DOM types
**TypeScript Version:** 2.8.3

**Search Terms:** 
- Intl.PluralRules
- Intl
- pluralrules

(searched all issues, the wiki, the code, google, and stackoverflow)

**Code**

```ts
const plurals = new Intl.PluralRules(); // error: Property 'PluralRules' does not exist on type 'typeof Intl'.
```

**Gist:**

`Intl.PluralRules` is at [Stage 4](https://github.com/tc39/proposal-intl-plural-rules) and is [supported by most browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/PluralRules). In addition, there is a TC39 polyfill that exists for older browsers. I'd expect this now to exist in the DOM type definitions.

**Expected behavior:** `Intl.PluralRules` to be defined.

**Actual behavior:** `Intl.PluralRules` isn't defined anywhere. (I actually even took it one further and ripgrepped the entire repository for `PluralRules` and there were 0 matches.)

**Playground Link:** N/A

**Related Issues:** None that I could locate.

### PR DESCRIPTION
**Pull request by dsifford**
This PR addresses the two separate issues discuessed in #23691. Namely:

1. There currently existed no type definitions for `Intl.PluralRules`.
2. The type definitions for `DateTimeFormat` were incorrectly associated with the global namespace, rather than in the `Intl` namespace.

I'd appreciate a code review on this if at all possible. Most of the changes made were based off of #15369, so there is a chance that I may have edited one or two things incorrectly.

Once the changes are found to be correct, if you could please let me know if/where you'd like tests placed for this, I'll go ahead and add those.

Thanks so much!

cc: @mhegazy 

Fixes #23691

### CODE REVIEW

File: lib/lib.es2018.d.ts [hunk 1]
```diff
@@ -20,4 +20,5 @@ and limitations under the License.
 
 /// <reference path="lib.es2017.d.ts" />
```
Comment (mhegazy) [inline lib/lib.es2018.d.ts]: do not update the lib files. they are generated as part of the build (`jake LKG`) and should not be edited manually.

Comment (mhegazy) [reply lib/lib.es2018.d.ts]: that applies to all files `lib\*`

Comment (dsifford) [reply lib/lib.es2018.d.ts]: Aha.. That makes sense. 

I had a feeling that was the case when editing the file.. 80,000+ lines! 

Would you like me to revert these?

Comment (mhegazy) [reply lib/lib.es2018.d.ts]: yes please.

**Comment by mhegazy** [comment]
thanks! wanna send a PR to update the docs as well at https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Compiler%20Options.md

**Comment by dsifford** [comment]
@mhegazy Sure!

**Pull request merged**